### ISSUE TITLE
cloudflare-ddns.service permission warning

### ISSUE THREAD

**Issue by highstrike** [Tachi107/cloudflare-ddns#24]
Hello,

Great tool! I installed it and it works great, however...
I wanted to see what the service outputs and noticed this warning
```
Mar 11 11:09:25 proxy (are-ddns)[995]: ConfigurationDirectory 'cloudflare-ddns' already exists but the mode is different. (File system: 755 ConfigurationDirectoryMode: 700)
```

full log below:
```
root@proxy:~# systemctl status cloudflare-ddns.service 
* cloudflare-ddns.service - Simple utility to dynamically change a DNS record
     Loaded: loaded (/lib/systemd/system/cloudflare-ddns.service; static)
     Active: inactive (dead) since Mon 2024-03-11 11:09:26 EET; 1min 9s ago
TriggeredBy: * cloudflare-ddns.timer
       Docs: man:cloudflare-ddns(1)
    Process: 995 ExecStart=/usr/bin/cloudflare-ddns (code=exited, status=0/SUCCESS)
   Main PID: 995 (code=exited, status=0/SUCCESS)
        CPU: 130ms

Mar 11 11:09:25 proxy systemd[1]: Starting cloudflare-ddns.service - Simple utility to dynamically change a DNS record...
Mar 11 11:09:25 proxy (are-ddns)[995]: ConfigurationDirectory 'cloudflare-ddns' already exists but the mode is different. (File system: 755 ConfigurationDirectoryMode: 700)
Mar 11 11:09:26 proxy cloudflare-ddns[995]: The A record is up to date
Mar 11 11:09:26 proxy cloudflare-ddns[995]: The AAAA record is up to date
Mar 11 11:09:26 proxy systemd[1]: cloudflare-ddns.service: Deactivated successfully.
Mar 11 11:09:26 proxy systemd[1]: Finished cloudflare-ddns.service - Simple utility to dynamically change a DNS record.
```

so then I went ahead and did `chmod 700 /etc/cloudflare-ddns/` and then restarted `systemctl restart cloudflare-ddns.service` and it failed with 
```
Job for cloudflare-ddns.service failed because the control process exited with error code.
See "systemctl status cloudflare-ddns.service" and "journalctl -xeu cloudflare-ddns.service" for details.

root@proxy:/etc# systemctl status cloudflare-ddns.service 
x cloudflare-ddns.service - Simple utility to dynamically change a DNS record
     Loaded: loaded (/lib/systemd/system/cloudflare-ddns.service; static)
     Active: failed (Result: exit-code) since Mon 2024-03-11 11:16:29 EET; 16s ago
TriggeredBy: * cloudflare-ddns.timer
       Docs: man:cloudflare-ddns(1)
    Process: 1059 ExecStart=/usr/bin/cloudflare-ddns (code=exited, status=1/FAILURE)
   Main PID: 1059 (code=exited, status=1/FAILURE)
        CPU: 22ms

Mar 11 11:16:29 proxy systemd[1]: Starting cloudflare-ddns.service - Simple utility to dynamically change a DNS record...
Mar 11 11:16:29 proxy cloudflare-ddns[1059]: Unable to open /etc/cloudflare-ddns/config.ini
Mar 11 11:16:29 proxy systemd[1]: cloudflare-ddns.service: Main process exited, code=exited, status=1/FAILURE
Mar 11 11:16:29 proxy systemd[1]: cloudflare-ddns.service: Failed with result 'exit-code'.
Mar 11 11:16:29 proxy systemd[1]: Failed to start cloudflare-ddns.service - Simple utility to dynamically change a DNS record.
```

🤷🏻

**Comment by highstrike** [comment]
It's perfect for my needs and all my subdomains that are CNAME (as they should be, lol)

Yes, I installed it via `apt install cloudflare-ddns` while as root on a debian 12 LXC running in Proxmox. The LXC is configured to autologin as root, and it doesn't even have a password, so yeah 😅

Here's `neofetch` if it helps:
```
       _,met$$$$$gg.          root@proxy 
    ,g$$$$$$$$$$$$$$$P.       ---------- 
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 12 (bookworm) x86_64 
 ,$$P'              `$$$.     Host: Z790M-ITX WiFi 
',$$P       ,ggs.     `$$b:   Kernel: 6.5.13-1-pve 
`d$$'     ,$P"'   .    $$$    Uptime: 10 hours, 29 mins 
 $$P      d$'     ,    $$P    Packages: 526 (dpkg) 
 $$:      $$.   -    ,d$$'    Shell: bash 5.2.15 
 $$;      Y$b._   _,d$P'      Terminal: /dev/tty1 
 Y$$.    `.`"Y$$$$P"'         CPU: Intel i5-14500T (2) @ 4.800GHz 
 `$$b      "-.__              GPU: Intel AlderLake-S GT1 
  `Y$$                        Memory: 133MiB / 1024MiB 
   `Y$$.
     `$$b.                                            
       `Y$$b.                                         
          `"Y$b._
              `"""
```

Indeed, my `/etc/cloudflare-ddns` directory is owned by root
`drwxr-xr-x  2 root root    4096 Mar 11 11:00 cloudflare-ddns`

Changing it via `chown -R cloudflare-ddns:cloudflare-ddns /etc/cloudflare-ddns` and adding back `chmod 700 /etc/cloudflare-ddns` resulted in the warning to dissapear, great success!

```
root@proxy:/etc# systemctl status cloudflare-ddns.service 

* cloudflare-ddns.service - Simple utility to dynamically change a DNS record
     Loaded: loaded (/lib/systemd/system/cloudflare-ddns.service; static)
     Active: inactive (dead) since Mon 2024-03-11 13:21:48 EET; 1s ago
TriggeredBy: * cloudflare-ddns.timer
       Docs: man:cloudflare-ddns(1)
    Process: 1793 ExecStart=/usr/bin/cloudflare-ddns (code=exited, status=0/SUCCESS)
   Main PID: 1793 (code=exited, status=0/SUCCESS)
        CPU: 130ms

Mar 11 13:21:47 proxy systemd[1]: Starting cloudflare-ddns.service - Simple utility to dynamically change a DNS record...
Mar 11 13:21:48 proxy cloudflare-ddns[1793]: The A record is up to date
Mar 11 13:21:48 proxy cloudflare-ddns[1793]: The AAAA record is up to date
Mar 11 13:21:48 proxy systemd[1]: cloudflare-ddns.service: Deactivated successfully.
Mar 11 13:21:48 proxy systemd[1]: Finished cloudflare-ddns.service - Simple utility to dynamically change a DNS record.
```

Thanks alot!

> In any case, that directory should automatically get the correct owner and permissions on install, so there's definitely something for me to fix :)

Glad I could help 😆

**Comment by Tachi107** [comment]
Fixed in 8eaa05a5bb5f28606bff675e95a1052a0a51c1dd by using a [`tmpfiles.d`](https://www.freedesktop.org/software/systemd/man/latest/tmpfiles.d.html) file!
